types/header.go

```diff
@@ -58,6 +58,11 @@ func (n Nonce) String() string {
 func (n Nonce) MarshalText() ([]byte, error) {
 	return []byte(n.String()), nil
 }
+func EncodeNonce(i uint64) Nonce {
+	var n Nonce
+	binary.BigEndian.PutUint64(n[:], i)
+	return n
+}
```

Comment:
Add a new line and consider to refactor it to a method.
```suggestion

func (n Nonce) EncodeNonce(i uint64) {
	binary.BigEndian.PutUint64(n[:], i)
}
```

Also invoke it in `func (h *Header) SetNonce(i uint64)`.